ActionPlans/Start-DlToO365GroupUpgradeChecks.ps1

```diff
@@ -0,0 +1,427 @@
+    $null = $TheObjectToConvertToHTML.Add($ConditionChildDGHTML)
+}
+#endregion Check if Distribution Group can't be upgraded because DlHasChildGroups
+
+#region Check if Distribution Group can't be upgraded because DlHasParentGroups
+[string]$SectionTitle = "Validating Distribution Group Parent Membership"
+[string]$Description = "Checking if Distribution Group can't be upgraded because it is a child group of another parent group"
+$ConditionParentDG=New-Object PSObject  
+try {
+    $alldgs=Get-DistributionGroup -ResultSize unlimited -ErrorAction Stop
+    $CurrentProperty = "Retrieving All DGs in the EXO directory"
+    $CurrentDescription = "Success"
+    write-log -Function "Retrieve All DGs" -Step $CurrentProperty -Description $CurrentDescription
+    
+}
+catch {
+    $CurrentProperty = "Retrieving All DGs in the EXO directory"
+    $CurrentDescription = "Failure"
+    write-log -Function "Retrieve All DGs" -Step $CurrentProperty -Description $CurrentDescription
+}  
+$parentdgcount=1
+foreach($parentdg in $alldgs)
+{
+    try {
+        $Pmembers = Get-DistributionGroupMember $($parentdg.Guid.ToString()) -ErrorAction Stop
+        $CurrentProperty = "Retrieving: $parentdg members"
+        $CurrentDescription = "Success"
+        write-log -Function "Retrieve Distrubtion Group membership" -Step $CurrentProperty -Description $CurrentDescription
+    }
+    catch {
+        $CurrentProperty = "Retrieving: $parentdg members"
+        $CurrentDescription = "Failure"
+        write-log -Function "Retrieve Distrubtion Group membership" -Step $CurrentProperty -Description $CurrentDescription
+    }
+
+foreach ($member in $Pmembers)
+{if ($member.alias -like $dg.alias)
+{
+    $ConditionParentDG|Add-Member -NotePropertyName "Parent Group$parentdgcount Alias" -NotePropertyValue $parentdg.Alias
+    $parentdgcount++
+}
+}
+}
+if($parentdgcount -le 1)
+{
+    $ConditionParentDG|Add-Member -NotePropertyName "Parent Group Alias" -NotePropertyValue "No parent groups found"
+    [PSCustomObject]$ConditionParentDGHTML = Prepare-ObjectForHTMLReport -SectionTitle $SectionTitle -SectionTitleColor "Green" -Description $Description -DataType "ArrayList" -EffectiveDataArrayList $ConditionParentDG -TableType "Table"
+    $null = $TheObjectToConvertToHTML.Add($ConditionParentDGHTML)
+}
+else {
+    [PSCustomObject]$ConditionParentDGHTML = Prepare-ObjectForHTMLReport -SectionTitle $SectionTitle -SectionTitleColor "Red" -Description $Description -DataType "ArrayList" -EffectiveDataArrayList $ConditionParentDG -TableType "Table"
+    $null = $TheObjectToConvertToHTML.Add($ConditionParentDGHTML)
+}
+#endregion Check if Distribution Group can't be upgraded because DlHasParentGroups
+
+#region Check if Distribution Group can't be upgraded because DlHasNonSupportedMemberTypes
+[string]$SectionTitle = "Validating Distribution Group Members Recipient Types"
+[string]$Description = "Checking if Distribution Group can't be upgraded because of unsupported member types"
```

Comment:
can you detailed what are unsupported member types?

Reply:
I've edited the comment before the check and description for HTML report.
#region Check if Distribution Group can't be upgraded because DlHasNonSupportedMemberTypes with RecipientTypeDetails other than UserMailbox, SharedMailbox, TeamMailbox, MailUser

$Description="Checking if Distribution Group can't be upgraded because DL contains member RecipientTypeDetails other than UserMailbox, SharedMailbox, TeamMailbox, MailUser"